app/models/miq_action.rb

```diff
@@ -848,6 +848,44 @@ def action_container_image_analyze(action, rec, inputs)
     rec.scan
   end
 
+  def action_container_image_prevent(action, rec, inputs)
+    unless rec.kind_of?(ContainerImage)
+      MiqPolicy.logger.error("MIQ(#{__method__}): Unable to perform action [#{action.description}],"\
+                             " object [#{rec.inspect}] is not a Container Image")
+      return
+    end
+
+    unless rec.try(:ext_management_system).kind_of?(ManageIQ::Providers::Openshift::ContainerManagerMixin)
+      MiqPolicy.logger.error("MIQ(#{__method__}): Unable to perform action [#{action.description}], "\
+                             "only applicable for OpenShift Providers")
+      return
+    end
+
+    unless rec.digest.present?
+      MiqPolicy.logger.error("MIQ(#{__method__}): Unable to perform action [#{action.description}], "\
+                             "ContainerImage is not linked with an OpenShift image")
+      return
+    end
+
+    if inputs[:synchronous]
```

Comment:
@moolitayer @Fryguy @gmcculloug @blomquisg  can anyone (after this patch) take the refactoring of the repeating:

```
if inputs[:synchronous]
  log_do_something
  do_something
else
  log_queuing_do_something
  queue(do_something)
end
```

It seems pretty common in this module.